src/join/composer.jl

```diff
@@ -55,8 +56,25 @@ struct DataFrameJoiner
                                         "when matchmissing == :error"))
                 end
             end
+        elseif matchmissing === :notequal
+            if kind in (:left, :semi, :anti)
+                dfr = dropmissing(dfr, right_on, view = true)
+                dfr_on = select(dfr, right_on)
+            elseif kind === :right
+                dfl = dropmissing(dfl, left_on, view = true)
+                dfl_on = select(dfl, left_on)
+            elseif kind === :inner
+                dfl = dropmissing(dfl, left_on, view = true)
```

Comment:
as a thought - for `:innejrojn` it is enough to `dropmissing` only for left or right table, as in the other table we will not find a match anyway. But maybe it is faster to perform `dropmissing` on both first as it lowers the cost of joining later. Have you checked this?

(probably this will not make a huge difference, so I am asking just to make sure we have considered this)

Reply:
Hmm, good catch. I will take a look at it tomorrow.

Reply:
As noted - it probably does not matter much, but it would be worth to double check. What you do now is a safe default (e.g. it might speed up things if data frame is sorted, but has `missing` sprinkled around as then faster join algorithm would be used).

If it does not matter much I would leave things as is. The reason is that in `innerjoin` dropping `missing`s might affect row order (as it will change which table - left or right is shorter), so in case it does not matter much the approach: we drop missings before joining is a good rule making sure that an advanced user will know what to expect in the output.

Reply:
I made some benchmarks with `innerjoin` and it's hard for me to decide on the best approach.
`innerjoin($p2, $v2, on = :Id => :PostId, makeunique = true, matchmissing = :notequal)`
I rerun experiments multiple times and this bump was consistent.

![memory](https://user-images.githubusercontent.com/9080845/120383660-6f184580-c325-11eb-97db-b6d0f9207df7.png)
![time](https://user-images.githubusercontent.com/9080845/120383663-6fb0dc00-c325-11eb-831a-9b6a77fb8a66.png)

Reply:
Interesting. As I have feared the best trade-off is not obvious (and probably depends on many factors) - let us keep "both drop" as it seems most robust. Thank you for checking this.

Reply:
Also, as you commented above, dropping only left/right changes the order of result and it took me some time to realize that I have to sort both dfs to compare them :D
I'll add comment in the code regarding possible gain in the future.